### ISSUE TITLE
`neovim` in bluefin-dx

### ISSUE THREAD

**Issue by Aman9das** [ublue-os/bluefin#918]
### Describe the package

I'd like to request the package `neovim` in bluefin-dx because it is one of the [most popular editors](https://survey.stackoverflow.co/2023/#most-popular-technologies-new-collab-tools) amongst developers.

Currently there is a lot of hassle with integrating neovim with the distro.
- overlay the package
- create an application shortcut `/usr/bin/gnome-terminal --full-screen --window -- nvim  %F`
Since ptyxis has an unnecessary headerbar when you want to focus on neovim only.

These steps can be automated. Thanks for your time.

### Image

![image](https://github.com/ublue-os/bluefin/assets/39594914/9733434a-1947-45f9-a9f9-372b1113f768)

**Comment by tepene** [comment]
I'm using the [bluefin-cli](https://universal-blue.discourse.group/docs?topic=39#introducing-bluefin-cli-7) for this. There you simply can run `brew install neovim`. And if you want to have it available on the host system you may add `distrobox-export --bin /home/linuxbrew/.linuxbrew/bin/nvim --export-path ~/.local/bin`.

**Comment by dylanmtaylor** [comment]
-1, there is no reason this has to be in the base image. As mentioned, you can use distrobox-export

**Comment by Aman9das** [comment]
The issue is for `bluefin-dx`. Distrobox exported neovim runs in the parent distrobox's environment.

If one wants to run neovim in another distrobox's environment, the `.local/bin/nvim` version takes precedence over the neovim installed in that distrobox. Effectively you can only have one dev environment regarding neovim if you export the binary.

 I like to have dedicated distroboxes per domain, due to this issue exporting the neovim binary is not viable.

TLDR: If `nvim` has been exported from distrobox `box1`. When invoking `nvim` from inside distrobox `box2`, it'll run from the `box1` environment even then. This is unwanted behaviour.

**Comment by dylanmtaylor** [comment]
Have you considered installing the nix package for it?

**Comment by castrojo** [comment]
Three possible solutions:

- -h per distrobox to keep the home directories scoped to each project/box
- [use a devcontainer](https://github.com/esensar/nvim-dev-container) which is the pattern we're defaulting to. 

I don't think layering it is that bad of a problem, set it once and then you probably will never have to deal with it. 

-1 to nix we don't really support that in Bluefin outside of using devbox.

**Comment by Aman9das** [comment]
Thanks for the devcontainer suggestion! I agree layering the package wouldn't be an issue.

Since vi and vim both are pre-installed. And Visual Studio Code is pre-installed. I continue to believe it'll be quite nice for bluefin-dx to include neovim too. But I think it's better to suggest this at fedora level itself

**Comment by castrojo** [comment]
Yeah, for the record shipping vscode on the image is a compromise, if the Flatpak was maintained by Microsoft then we'd move over it it.
